src/header.rs

```diff
@@ -856,8 +856,8 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
     // when we add support for it.
     let width = fi.width - 1;
     let height = fi.height - 1;
-    let width_bits = 32 - (width as u32).leading_zeros();
-    let height_bits = 32 - (height as u32).leading_zeros();
+    let width_bits = log_in_base_2(fi.width as u32) as u32 + 1;
+    let height_bits = log_in_base_2(fi.height as u32) as u32 + 1;
```

Comment:
As mentioned on IRC, this regresses the change from #2510 (that fixed issue #2508)
```
[trace_headers @ 0000020b1aea7480] 46          frame_width_bits_minus_1                                 0110 = 6
[trace_headers @ 0000020b1aea7480] 50          frame_height_bits_minus_1                                0110 = 6
[trace_headers @ 0000020b1aea7480] 54          max_frame_width_minus_1                               0111111 = 63
[trace_headers @ 0000020b1aea7480] 61          max_frame_height_minus_1                              0111111 = 63
```